inc/template.php

```diff
@@ -1,35 +1,69 @@
 <?php
+/**
+ * Templating functions.
+ */
 
 namespace HtmxPress\Template;
 
 use const HtmxPress\Endpoint\HTMX_ENDPOINT;
 
-function bootstrap() {
+/**
+ * Bootstrap.
+ *
+ * @return void
+ */
+function bootstrap() : void {
 	add_action( 'template_redirect', __NAMESPACE__ . '\\render' );
 }
 
-function render() {
+/**
+ * Render a template.
+ *
+ * @return void
+ */
+function render() : void {
 	global $wp_query;
 	if ( ! isset( $wp_query->query_vars[ HTMX_ENDPOINT ] ) ) {
 		return;
 	}
 
 	// POST nonce protection
-	if ( $_SERVER["REQUEST_METHOD"] === 'POST' && ! is_nonced() ) {
+	if ( $_SERVER['REQUEST_METHOD'] === 'POST' && ! is_nonced() ) {
 		return;
 	}
 	$template_name = get_template_name();
-	load_template_or_404( $template_name );
+	$template_loaded = load_htmx_template( $template_name );
+	maybe_set_404( $template_loaded );
 	exit;
 }
 
+/**
+ * Set 404 status if the template wasn't loaded.
+ *
+ * @param bool $template_loaded Whether a template was loaded.
+ *
+ * @return void
+ */
+function maybe_set_404( bool $template_loaded ) : void {
+	global $wp_query;
+	if ( ! $template_loaded ) {
+		$wp_query->set_404();
+		status_header( 404 );
+	}
+}
+
+/**
+ * Check if the request is nonced.
+ *
+ * @return bool
+ */
 function is_nonced() : bool {
 	$nonce = null;
 
 	if ( isset( $_REQUEST['_wpnonce'] ) ) {
-		$nonce = $_REQUEST['_wpnonce'];
+		$nonce = sanitize_text_field( wp_unslash( $_REQUEST['_wpnonce'] ) );
 	} elseif ( isset( $_SERVER['HTTP_X_WP_NONCE'] ) ) {
-		$nonce = $_SERVER['HTTP_X_WP_NONCE'];
+		$nonce = sanitize_text_field( wp_unslash( $_SERVER['HTTP_X_WP_NONCE'] ) );
 	}
```

Comment:
I would say it's best practice to sanitize first, then check the values (the `isset()` call). I think ... I could be wrong here, so don't quote me ... there are a few instances where the `sanitize_text_field()` that would cause the `isset()` to return false, specifically if it removed invalid UTF-8.

Quick question, and forgive my ignorance, is this meant to support both `GET` and `POST` requests with `$_REQUEST['_wpnonce']`?